kuka_rsi_hw_interface/src/kuka_hardware_interface_node.cpp

```diff
@@ -50,6 +50,58 @@ int main(int argc, char** argv)
 
   ros::NodeHandle nh;
 
+  std::ifstream realtime_file("/sys/kernel/realtime", std::ios::in);
+  bool has_realtime = false;
+  if (realtime_file.is_open())
+  {
+    realtime_file >> has_realtime;
+  }
+  if (has_realtime)
+  {
+    const int max_thread_priority = sched_get_priority_max(SCHED_FIFO);
+    if (max_thread_priority != -1)
+    {
+      // We'll operate on the currently running thread.
+      pthread_t this_thread = pthread_self();
+
+      // struct sched_param is used to store the scheduling priority
+      struct sched_param params;
+
+      // We'll set the priority to the maximum.
+      params.sched_priority = max_thread_priority;
+
+      int ret = pthread_setschedparam(this_thread, SCHED_FIFO, &params);
+      if (ret != 0)
+      {
+        ROS_ERROR_STREAM("Unsuccessful in setting main thread realtime priority. Error code: " << ret);
+      }
+      // Now verify the change in thread priority
+      int policy = 0;
+      ret = pthread_getschedparam(this_thread, &policy, &params);
+      if (ret != 0)
+      {
+        std::cout << "Couldn't retrieve real-time scheduling paramers" << std::endl;
```

Comment:
Shouldn't this be `ROS_ERROR_STREAM` as well?

Reply:
Yes. Adjusted in [06f0ab7](https://github.com/ros-industrial/kuka_experimental/pull/257/commits/06f0ab7071e101ba005ab343d92d1439cdadf8a2)